frontend/src/models/affinities.js

```diff
@@ -0,0 +1,74 @@
+export const affinityEvents = {
+  announcement: "announcement",
+  queen_of_clubs: "queen_of_clubs",
+  played_card: "played_card"
+};
+
+export class Affinities {
+  constructor(me, players = []) {
+    this.me = me;
+    this.setPlayers(players);
+  }
+
+  setPlayers(players) {
+    this.players = [];
+    players.forEach(player => {
+      if (this.me.id !== player.id) {
+        this.players.push({ player, affinity: 0 });
+      }
+    });
+  }
+
+  declaresParty(player) {
+    if (this._isMyPartyMember(player)) {
+      this.players.forEach(element => {
+        this.setPlayerAffinityByParty(element.player);
+      });
+    } else {
+      this.setPlayerAffinityByParty(player);
+    }
+  }
+
+  getPlayerAffinityValue(player) {
+    return !this._isMe(player)
```

Comment:
I didn't do this, because I wanted to sum up affinites. Affinities should always sum up to -1.
Having another element in the array could cause some issues.

We can use this to calculate with whom we believe we're playing, if we initialize all players with -0.3.
I believe this might be a good way of having the AI to choose between the lesser evil 😄

Reply:
Oh, thought of this: -1 won't work for "Stiller Solo" and since this isn't announced...